### PR TITLE
Fixes for ChatInterface Examples when additional inputs are provided

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
- Fixes issue where examples were being shown twice, in chatbot and under it
- Now only uses `gr.Examples` when `additional_inputs` are provided in the actual example

Closes: #9732

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/chat_interface.py [hunk 1]
```diff
@@ -717,6 +728,13 @@ def _process_example(
                 ]
         return result
 
+    def _additional_inputs_in_examples(self):
```
Comment (dawoodkhan82) [inline gradio/chat_interface.py:731]: Is there a better way to detect if additional inputs are in the provided example?

Comment (abidlabs) [reply gradio/chat_interface.py:731]: We'll need to traverse the `examples` array and check to see if the values for the additional_examples are `None`.

See https://github.com/gradio-app/gradio/blob/c45b46652d49e0a8a5d8217bbbc6ddbe4b4e91f0/gradio/helpers.py#L221 where we do something similar

**Comment by aliabid94** [comment]
Clicking on an example doesn't seem to do anything anymore? 

Recording 2024-10-23 at 14 31 26

**Comment by dawoodkhan82** [comment]
Will fix

File: gradio/chat_interface.py [hunk 2]
```diff
@@ -717,6 +728,20 @@ def _process_example(
...
+        if self.examples is not None:
+            for example in self.examples:
+                for idx, example_for_input in enumerate(example):
+                    if example_for_input is not None:
+                        try:
+                            if idx > 0:
+                                return True
+                            else:
+                                continue
+                        except IndexError:
```
Comment (freddyaboulton) [inline gradio/chat_interface.py:741]: I don't think we need the `IndexError` check. What code in the loop would raise it?

**Review by freddyaboulton — APPROVED** [review]
Looks good @dawoodkhan82 !

**Pull request merged**